jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java

```diff
@@ -102,35 +102,41 @@ public TestEnvironmentRestController(TestEnvironmentService testEnvironmentServi
 
         if (!testEnvService.existsWithSessionId(envId, sessionId))
             throw new TestEnvironmentSessionNotFoundException(envId, sessionId);
+        validateTestEnv(testEnv);
+        TestEnvironmentEntity oldEnv = testEnvService.read(envId);
 
         testEnv.setEnvironmentId(envId);
         TestEnvironmentEntity updated = testEnvService.update(testEnv);
-        response.addCookie(getSessionCookie(updated));
         if (updated.getStatus() == PENDING) {
-            getTestSuiteNameToExecute(updated).ifPresent(testSuiteName -> setNextConfigToExecuteHeader(response, testSuiteName));
+            if (oldEnv.getStatus() == RUNNING)
+                testExecutionService.finishExecution(envId, oldEnv.getRunningLoadScenario().getLoadScenarioId());
```

Comment:
- previous load scenario is over
- master is reporting pending state
- we should return next load scenario to execute

my question: why if master was running smth before and now is pending, we are not returning scenario to execute = skipping one communication cycle